### ISSUE TITLE
V**(1/3) is NaN for cells with negative determinant

### ISSUE THREAD

**Issue by janosh** [WillBaldwin0/graph_electrostatics#6]
`slabs.py:152` and `:241` take a cube root of the signed cell volume:

```python
Ls = torch.pow(volumes, 0.333333)
```

Any left-handed cell makes `Ls` NaN, and `MonopoleDipoleCorrectionBlock` and `CorrectivePotentialBlock` NaN with it. A negative determinant is legal in ASE and pymatgen, and swapping two lattice vectors produces one. `L` is used as a length scale, so `|V|` looks like the intent.

```python
import torch
from graph_longrange.slabs import MonopoleDipoleCorrectionBlock

block = MonopoleDipoleCorrectionBlock(1)
charges = torch.tensor([[0.4, 0.1, 0.0, -0.2], [-0.4, 0.0, 0.1, 0.3]])
positions = torch.tensor([[0.1, 0.0, 0.0], [1.5, 0.3, 0.2]])
batch = torch.zeros(2, dtype=torch.long)
for volume in (262.144, -262.144):
    print(volume, block(charges, positions, torch.tensor([volume]), batch))
```

```
262.144 tensor([0.0244])
-262.144 tensor([nan])
```

In a fully periodic `PolarMACE` run the correction is masked out by `torch.where` in `energy.py`, so the energy stays finite. The forces come back NaN and the stress is reported as exactly zero. The result looks like output rather than a failure.

The exponents are also truncated and disagree: `0.333333` at line 152, `0.3333` at line 241. The latter is a 1.5e-4 to 3.4e-4 relative error in `L` for cells of 10^2 to 10^4 A^3. `1/3` is free.
